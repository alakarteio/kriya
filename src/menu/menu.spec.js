/* eslint-disable import/no-extraneous-dependencies, react/jsx-filename-extension */
/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import Menu from './menu'

const snap = (props) => {
  const component = renderer.create(
    <Menu {...props} />,
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
}

describe('common/Menu', () => {
  it('should have a default behaviour', () => snap({}))
  it('should add custom className', () => snap({ className: 'custom' }))
  it('should add custom style', () => snap({ style: { backgroundColor: 'red' } }))
  it('should add a header', () => snap({ header: <div>header</div> }))
  it('should add a footer', () => snap({ footer: <div>footer</div> }))
  it('should add items', () => snap({ children: [<div key={1}>1</div>, <div key={2}>2</div>] }))
})

/* eslint-enable import/no-extraneous-dependencies, react/jsx-filename-extension */
