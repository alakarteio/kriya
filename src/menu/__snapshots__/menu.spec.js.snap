// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Menu should add a footer 1`] = `
<div
  className="menu"
  style={Object {}}
>
  <div
    className="sep"
  />
  <div
    className="items"
  />
  <div
    className="footer"
  >
    <div>
      footer
    </div>
  </div>
</div>
`;

exports[`common/Menu should add a header 1`] = `
<div
  className="menu"
  style={Object {}}
>
  <div>
    header
  </div>
  <div
    className="sep"
  />
  <div
    className="items"
  />
  <div
    className="footer"
  />
</div>
`;

exports[`common/Menu should add custom className 1`] = `
<div
  className="menu custom"
  style={Object {}}
>
  <div
    className="sep"
  />
  <div
    className="items"
  />
  <div
    className="footer"
  />
</div>
`;

exports[`common/Menu should add custom style 1`] = `
<div
  className="menu"
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  <div
    className="sep"
  />
  <div
    className="items"
  />
  <div
    className="footer"
  />
</div>
`;

exports[`common/Menu should add items 1`] = `
<div
  className="menu"
  style={Object {}}
>
  <div
    className="sep"
  />
  <div
    className="items"
  >
    <div>
      1
    </div>
    <div>
      2
    </div>
  </div>
  <div
    className="footer"
  />
</div>
`;

exports[`common/Menu should have a default behaviour 1`] = `
<div
  className="menu"
  style={Object {}}
>
  <div
    className="sep"
  />
  <div
    className="items"
  />
  <div
    className="footer"
  />
</div>
`;
